### PR TITLE
Remove old testsuite

### PR DESCRIPTION
**Pull request by dgdavid**
## Problem

The legacy test suite, which makes testing by output to log, is a blocker for the evolution of YaST and is causing compatibility problems with Ruby 2.7 failing to build in [Staging:J](https://build.opensuse.org/project/show/openSUSE:Factory:Staging:J). 

## Solution

The YaST team has been trying to migrate as much as possible all legacy YaST module test suites to RSpec. However, among others this module is mainly written in Perl and we didn't find a good Perl-mocking tool to make that it less fragile and more useful.

Looking at the history of changes it can be seen that those tests (and the related code) have been mostly untouched for ages, which helps to make the decision of getting rid of them and move on.

So, this PR completely **deletes the old test suite without any replacement**. Of course, it can be checked any time whether needed thanks to the version control history. Ideally, every new change from now on should be tested with RSpec, although it still needed either, to migrate the existing Perl code to Ruby or find out the way to mock it properly in order to use RSpec for writing the tests.

## Other changes

`BuildRequire`s has been also updated
* adding the necessary dependencies to run the remaining test suite at the build time, and
* removing not needed samba packages.

Thanks to @mvidner for [the clue](#issuecomment-634756266)

---

See also https://github.com/yast/yast-samba-client/pull/84

### CODE REVIEW

**Comment by dgdavid** [comment]
Hey @yast/samba-team, friendly ping here :)

Although I requested review from other YaST team members, I'd like to have a review from you too :)

Thanks in advance!

File: package/yast2-samba-server.spec [hunk 1]
```diff
@@ -30,7 +30,6 @@ Source0:        %{name}-%{version}.tar.bz2
 BuildRequires:  libsmbclient-devel
 BuildRequires:  perl-Crypt-SmbHash
 BuildRequires:  perl-X500-DN
-BuildRequires:  perl-XML-Writer
```
Comment (ancorgs) [inline package/yast2-samba-server.spec:33]: I miss dropping the BuildRequires on yast2-testsuite (looks like we always forget one of the two).

Comment (dgdavid) [reply package/yast2-samba-server.spec:33]: Weird. I would swear I deleted it... 

Anyway, fixed :)

**Comment by mvidner** [comment]
The remaining tests in `test/` are only run in Travis, not at RPM build time.
Should we BuildRequire rspec and run it?

Also, we BuildRequire several samba packages, probably unnecessarily.

**Comment by dmulder** [comment]
lgtm

**Pull request merged**